client/components/map.jsx

```diff
@@ -1,18 +1,122 @@
 import React from 'react';
-import { RenderMap } from './render-map';
+import { GoogleMap, LoadScript, Marker, InfoWindow, DirectionsRenderer } from '@react-google-maps/api';
 import AppContext from '../lib/app-context';
 
-export default class HomeMap extends React.Component {
+const key = process.env.GOOGLE_MAPS_API_KEY;
+
+export default class Map extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
       lat: 33.63487,
       lng: -117.74045,
-      gymInfo: []
+      gymInfo: [],
+      currentPosition: null,
+      destination: null,
+      trip: {
+        distance: null,
+        duration: null,
+        directions: null
+      },
+      clickedGym: null
     };
+    this.handleClick = this.handleClick.bind(this);
+    this.renderMarkers = this.renderMarkers.bind(this);
+    this.getDirections = this.getDirections.bind(this);
+  }
+
+  handleClick(event, gym) {
+    this.setState({
+      clickedGym: gym,
+      destination: {
+        lat: event.latLng.lat(),
+        lng: event.latLng.lng()
+      }
+    });
+  }
+
+  getDirections() {
+    const directionsService = new window.google.maps.DirectionsService();
+
+    const origin = this.state.currentPosition;
+    const destination = this.state.destination;
+
+    if (origin !== null && destination !== null) {
+      directionsService.route(
+        {
+          origin,
+          destination,
+          travelMode: window.google.maps.TravelMode.DRIVING
+        },
+        (result, status) => {
+          this.setState({
+            trip: {
+              distance: result.routes[0].legs[0].distance,
+              duration: result.routes[0].legs[0].duration,
+              directions: result.routes[0].legs[0].steps
+            }
+          });
+          if (status === window.google.maps.DirectionsStatus.OK && result !== null) {
+            this.setState({
+              directions: result
+            });
+          } else {
+            console.error(`error fetching directions ${result}`);
+          }
+
+        }
+      );
+    }
+  }
+
+  renderMarkers(gyms) {
+    return gyms.map((coordinates, i) => {
+      const coords = {
+        lat: gyms[i].coordinates.latitude,
+        lng: gyms[i].coordinates.longitude
+      };
+      const title = gyms[i].name;
+      return (
+      <Marker
```

Comment:
There's something a little weird going on in this `.map()`. instead of using `array[i]`, you typically use the first parameter of your callback (because it is `array[i]`)